File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/promises/extensions/PromiseMessenger.java
package com.lasthopesoftware.bluewater.shared.promises.extensions;

import com.namehillsoftware.handoff.Messenger;
import com.namehillsoftware.handoff.promises.Promise;

public class PromiseMessenger<Resolution> extends Promise<Resolution> implements Messenger<Resolution> {
	@Override
	public void sendResolution(Resolution resolution) {
		resolve(resolution);
	}

	@Override
	public void sendRejection(Throwable error) {
		reject(error);
	}

	@Override
	public void cancellationRequested(Runnable response) {
		respondToCancellation(response);
	}
}

File: projectBlueWater/src/main/java/com/lasthopesoftware/bluewater/shared/promises/extensions/PromiseMessenger.kt
package com.lasthopesoftware.bluewater.shared.promises.extensions

import com.namehillsoftware.handoff.Messenger
import com.namehillsoftware.handoff.promises.Promise

class PromiseMessenger<Resolution> : Promise<Resolution>(), Messenger<Resolution> {
    override fun sendResolution(resolution: Resolution) {
        resolve(resolution)
    }

    override fun sendRejection(error: Throwable) {
        reject(error)
    }

    override fun cancellationRequested(response: Runnable) {
        respondToCancellation(response)
    }
}
